datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import json
+import logging
+import os
+import pathlib
+import typing
+
+import pandas as pd
+import requests
+from google.cloud import storage
+
+
+def main(
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+    headers: typing.List[str],
+    rename_mappings: dict,
+    pipeline_name: str,
+) -> None:
+
+    logging.info(
+        f"FDIC{pipeline_name} process started at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+    logging.info("Creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+
+    logging.info(f"Downloading file from {source_url}...")
+    download_file(source_url, source_file)
+
+    logging.info(f"Opening file {source_file}...")
+    df = pd.read_csv(str(source_file))
+
+    logging.info(f"Transforming {source_file}... ")
+
+    logging.info("Renaming Columns...")
+    rename_headers(df, rename_mappings)
+
+    if pipeline_name == "locations":
+        logging.info("Replacing bool values...")
+        replace_bool_list = [
+            "main_office",
+            "cbsa_division_flag",
+            "cbsa_metro_flag",
+            "cbsa_micro_flag",
+            "csa_flag",
+        ]
+        replace_bool(replace_bool_list, df)
+        df["cbsa_division_fips_code"] = df["cbsa_division_fips_code"].astype(
+            "Int64", errors="ignore"
+        )
+        logging.info("Replacing date values...")
+        format_date_list = ["date_established", "last_updated"]
+        format_date(format_date_list, df)
+        logging.info("Replacing with null values...")
+        df["cbsa_metro_fips_code"] = df["cbsa_metro_fips_code"].replace(0, "NULL")
+    else:
+        logging.info("Replacing bool values...")
+        replace_bool_list = [
```

Comment:
Please move to pipeline.yaml and incorporate as a variable

Reply:
done